syft/pkg/cataloger/ai/parse_gguf_model.go

```diff
@@ -0,0 +1,127 @@
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	gguf_parser "github.com/gpustack/gguf-parser-go"
+
+	"github.com/anchore/syft/internal"
+	"github.com/anchore/syft/internal/unknown"
+	"github.com/anchore/syft/syft/artifact"
+	"github.com/anchore/syft/syft/file"
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/anchore/syft/syft/pkg/cataloger/generic"
+)
+
+const unknownGGUFData = "unknown"
+
+// parseGGUFModel parses a GGUF model file and returns the discovered package.
+// This implementation only reads the header portion of the file, not the entire model.
+func parseGGUFModel(_ context.Context, _ file.Resolver, _ *generic.Environment, reader file.LocationReadCloser) ([]pkg.Package, []artifact.Relationship, error) {
+	defer internal.CloseAndLogError(reader, reader.Path())
+
+	// Read only the header portion (not the entire file)
+	headerReader := &ggufHeaderReader{reader: reader}
+	headerData, err := headerReader.readHeader()
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to read GGUF header: %w", err)
+	}
+
+	// Create a temporary file with just the header for the library to parse
+	// The library requires a file path, so we create a minimal temp file
+	tempFile, err := os.CreateTemp("", "syft-gguf-header-*.gguf")
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to create temp file: %w", err)
+	}
+	tempPath := tempFile.Name()
+	defer os.Remove(tempPath)
+
+	// Write header data to temp file
+	if _, err := tempFile.Write(headerData); err != nil {
+		tempFile.Close()
+		return nil, nil, fmt.Errorf("failed to write header to temp file: %w", err)
+	}
+	tempFile.Close()
+
+	// Parse using gguf-parser-go with options to skip unnecessary data
+	ggufFile, err := gguf_parser.ParseGGUFFile(tempPath,
+		gguf_parser.SkipLargeMetadata(),
+	)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to parse GGUF file: %w", err)
+	}
+
+	// Extract metadata
+	metadata := ggufFile.Metadata()
+
+	// Convert to syft metadata structure
+	syftMetadata := &pkg.GGUFFileHeader{
+		ModelFormat:     "gguf",
+		ModelName:       metadata.Name,
+		ModelVersion:    extractVersion(ggufFile.Header.MetadataKV),
+		License:         metadata.License,
+		Architecture:    metadata.Architecture,
+		Quantization:    metadata.FileTypeDescriptor,
+		Parameters:      uint64(metadata.Parameters),
+		GGUFVersion:     uint32(ggufFile.Header.Version),
+		TensorCount:     ggufFile.Header.TensorCount,
+		Header:          convertGGUFMetadataKVs(ggufFile.Header.MetadataKV),
+		TruncatedHeader: false, // We read the full header
+		Hash:            "",    // Will be computed in newGGUFPackage
+	}
+
+	// If model name is not in metadata, use filename
+	if syftMetadata.ModelName == "" {
+		syftMetadata.ModelName = extractModelNameFromPath(reader.Path())
+	}
+
+	// If version is still unknown, try to infer from name
+	if syftMetadata.ModelVersion == unknownGGUFData {
+		syftMetadata.ModelVersion = extractVersionFromName(syftMetadata.ModelName)
+	}
```

Comment:
these could be persisted in a local var and passed into the package contructor instead of being on the metadata struct